src/flow_solver/flow_solver_cases/periodic_turbulence.cpp

```diff
@@ -198,6 +199,103 @@ double PeriodicTurbulence<dim, nstate>::get_deviatoric_strain_rate_tensor_based_
+            if(ishape == 0)
+                soln_coeff[istate].resize(n_shape_fns);
+            soln_coeff[istate][ishape] = dg->solution(dofs_indices[idof]);
+        }
+
+        std::array<std::vector<double>,nstate> soln_at_q;
+        for(int istate=0; istate<nstate; istate++){
+            soln_at_q[istate].resize(n_quad_pts);
+            soln_basis.matrix_vector_mult_1D(soln_coeff[istate], soln_at_q[istate],
+                                             soln_basis.oneD_vol_operator);
+        }
+        std::array<std::vector<double>,nstate> entropy_var_at_q;
+        for(unsigned int iquad=0; iquad<n_quad_pts; iquad++){
+            std::array<double,nstate> soln_state;
+            for(int istate=0; istate<nstate; istate++){
+                soln_state[istate] = soln_at_q[istate][iquad];
+            }
+
+            std::array<double,nstate> entropy_var;
+            const double density = soln_state[0];
+            dealii::Tensor<1,dim,double> vel;
+            double vel2 = 0.0;
+            for(int idim=0; idim<dim; idim++){
+                vel[idim] = soln_state[idim+1]/soln_state[0];
+                vel2 += vel[idim]*vel[idim];
+            }
+            const double pressure = 0.4*(soln_state[nstate-1] - 0.5*density*vel2);
+            const double entropy = log(pressure) - 1.4 * log(density);
+            
+            //pcout << pressure << " " << entropy << std::endl;        
+            entropy_var[0] = (1.4-entropy)/0.4 - 0.5 * density / pressure * vel2;
+            for(int idim=0; idim<dim; idim++){
+                entropy_var[idim+1] = soln_state[idim+1] / pressure;
+            }
+            entropy_var[nstate-1] = - density / pressure;
+            //pcout << entropy_var[0] << " ";
+
+            for(int istate=0; istate<nstate; istate++){
+                if(iquad==0)
+                    entropy_var_at_q[istate].resize(n_quad_pts);
+                entropy_var_at_q[istate][iquad] = entropy_var[istate];
+            }
+        }
+        for(int istate=0; istate<nstate; istate++){
+            //Projected vector of entropy variables.
+            std::vector<double> entropy_var_hat(n_shape_fns);
+            vol_projection.matrix_vector_mult_1D(entropy_var_at_q[istate], entropy_var_hat,
+                                                 vol_projection.oneD_vol_operator);
+                                                
+            for(unsigned int ishape=0; ishape<n_shape_fns; ishape++){
+                const unsigned int idof = istate * n_shape_fns + ishape;
+                entropy_var_hat_global[dofs_indices[idof]] = entropy_var_hat[ishape];
+            }
+            //this->pcout << entropy_var_hat_global[0] << " ";
+        }
+    }
+
+    double entropy = entropy_var_hat_global * mass_matrix_times_solution;
+    //double entropy_mpi = (dealii::Utilities::MPI::sum(entropy, this->mpi_communicator));
```

Comment:
should be uncommented and returning `entropy_mpi` ?

Reply:
I believe that the dot product accounts for MPI. The `deal.ii` [documentation](https://www.dealii.org/current/doxygen/deal.II/group__Vectors.html#ga829281a9ccc9f1cf9720ee5a535efd21) is a little unclear, but when I uncomment that line, MPI with 4 processes returns `Num. Entropy: -9596.309109184054` and serial returns `Num. Entropy: -2399.077277296014` (exactly 1/4 x the MPI). I'll remove that commented line.

Reply:
Ah okay, maybe just leave a comment in the code that the dot product accounts for the MPI

Reply:
For sure. It's a little unclear, good to make a note of that in the code! Just pushed an update with a comment.